src/pal/src/loader/module.cpp

```diff
@@ -1339,29 +1339,33 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
     // See if file can be dlopen()ed; this should work even if it's already loaded
     {
         // See GetProcAddress for an explanation why we leave the PAL.
-        PAL_LeaveHolder holder;
-        dl_handle = dlopen(shortAsciiName, RTLD_LAZY | RTLD_NOLOAD); 
-        if (!dl_handle)
+        PAL_LeaveHolder holder; 
+        
+        // P/Invokes are often declared with variations on the actual library name.
+        // For example, it's common to leave off the extension/suffix of the library
+        // even if it has one, or to leave off a prefix like "lib" even if it has one
+        // (both of these are done typically to smooth over cross-platform differences). 
+        // We try to dlopen with such variations on the original.
+        const char* const formatStrings[4] = // used with args: PAL_SHLIB_PREFIX, shortAsciiName, PAL_SHLIB_SUFFIX
         {
-            dl_handle = dlopen(shortAsciiName, RTLD_LAZY);
-        }
-
-        // P/Invoke calls are often defined without an extension in the name of the 
-        // target library. So if we failed to load the specified library, try adding
-        // a proper extension and load the library again.
-        if (!dl_handle)
+            "%s%s%s",     // prefix+name+suffix
+            "%.0s%s%.0s", // name
```

Comment:
Thank you. Sounds good. As I said, I’m OK with the proposed order. I also don’t have any objections against promoting the name without any prefix/suffix but I’m not sure whether PAL is the right place for doing that. I would love to see PAL (as the lowest layer of the runtime stack) to be policy free. It is not going to be easy for an app to change it if we get it wrong.

In any case, Steve, please feel to merge your change.